components/global/at_mentions_button/at_mentions_button.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Tooltip} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import IconButton from '@mattermost/compass-components/components/icon-button';
+
+import OverlayTrigger from 'components/overlay_trigger';
+import Constants, {RHSStates} from 'utils/constants';
+
+type Props = {
+    rhsState: typeof RHSStates[keyof typeof RHSStates] | null;
+    isRhsOpen?: boolean;
+    actions: {
+        showMentions: () => void;
+        closeRightHandSide: () => void;
+    };
+};
+
+const AtMentionsButton = (props: Props): JSX.Element => {
+    const {rhsState, isRhsOpen, actions: {closeRightHandSide, showMentions}} = props;
```

Comment:
haha I think this is definitely one of those things that are simple but harder to agree on, I much prefer destructuring props all together but since there isn't much repetition in these components I can bring it back.

Reply:
I personall agree with the destructuring approach, although it could possibly be done directly in the function parameter list instead of a new const statement.

```javasctipt
const MyFunction = ({prop1, prop2}: Props): JSX.Element => {...}
```

Modern IDE's make it very easy to determine the origin of a variable, though I do understand the momentary uncertainty! The challenge is that we are going to have many different opinions on syntax style preferences and so would need to agree on a standardized approach. I personally like the idea of adopting an existing style guide that is considered by the community in general to be a decent standard to follow vs trying to define one ourselves.